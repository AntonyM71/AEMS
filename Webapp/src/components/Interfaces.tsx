import { AthleteInfo } from "./roles/scribe/InfoBar"

export interface OverlayControlState {
	// flags to show/hide different components
	showTimer: boolean
	showImageCard: boolean
	showHeatSummary: boolean
	showLiveRunScore: boolean
	showPhaseResults: boolean
	showEventTitle: boolean
	// data
	selectedCompetition: string
	selectedEvent: string
	selectedPhase: string
	selectedHeat: string

	selectedAthlete: AthleteInfo | undefined
	selectedRun: number
}

export const defaultOverlayControllerState: OverlayControlState = {
	showTimer: false,
	showImageCard: true,
	showHeatSummary: false,
	showEventTitle: false,
	showLiveRunScore: false,
	showPhaseResults: false,
	selectedCompetition: "",
	selectedEvent: "",
	selectedPhase: "",
	selectedHeat: "",
	selectedAthlete: undefined,
	selectedRun: 0
}
