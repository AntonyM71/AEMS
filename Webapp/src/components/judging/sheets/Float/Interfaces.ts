export interface BonusPropsType {
	bonus: bonusesType
	addScoredMove: addScoredMoveType
	addScoredBonus: addScoredBonusType
}
export interface bonusesType {
	id: string
	name: string
	shortName: string
}

export interface movesType {
	id: string
	name: string
	direction: AvailableMoveDirections
	score: number
}

export type AvailableMoveDirections = "SINGLE" | "LR" | "FB" | "LRFB"
export type directionType = "L" | "R" | "F" | "B" | "LF" | "RB" | "A"
export type removeScoredMoveType = (id: string) => void
export interface scoredMovesType {
	id: string
	timestamp: Date
	moveId: string
	direction: directionType
	bonuses: scoredBonusType[]
	status: string
}
export interface scoredBonusType {
	id: string
	timestamp: Date
	bonusId: string
}

export interface stateType {
	currentMoveId: string
	scoredMoves: scoredMovesType[]
	scores: {
		heat: number
		run: number
	}
}
export type addScoredMoveType = (id: string, direction: directionType) => void
export type addScoredBonusType = (id: string, moveId: string) => void

export interface MovePropsType {
	move: movesType
	addScoredMove: addScoredMoveType
	addScoredBonus: addScoredBonusType
}
