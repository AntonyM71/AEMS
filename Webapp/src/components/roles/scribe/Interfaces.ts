export interface BonusPropsType {
	bonus: bonusesType
	addScoredMove: addScoredMoveType
	addScoredBonus: addScoredBonusType
}
export interface bonusesType {
	id: string
	name: string
	shortName: string
}

export interface movesType {
	id: string
	name: string
	direction: AvailableMoveDirections
	fl_score: number
	rb_score: number
}

export type AvailableMoveDirections = "S" | "LR" | "FB"
export type directionType = frontLeftDirections | rightBackDirections | "S"
export type frontLeftDirections = "L" | "F"
export type rightBackDirections = "R" | "B"
export type removeScoredMoveType = (id: string) => void

export interface scoredMovesType {
	id: string

	moveId: string
	direction: directionType
}
export interface scoredBonusType {
	id: string
	moveId: string
	bonusId: string
}

export interface stateType {
	currentMoveId: string
	scoredMoves: scoredMovesType[]
	scores: {
		heat: number
		run: number
	}
}
export type addScoredMoveType = (id: string, direction: directionType) => void
export type addScoredBonusType = (id: string, moveId: string) => void

export interface MovePropsType {
	move: movesType
}
