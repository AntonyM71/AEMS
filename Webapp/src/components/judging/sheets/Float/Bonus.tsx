import Button from "@mui/material/Button"
import Paper from "@mui/material/Paper"
import Typography from "@mui/material/Typography"
import React from "react"

import { BonusPropsType } from "./Interfaces"

export const Bonus = (props: BonusPropsType) => (
		<Paper >
			<Typography>{props.bonus.name}</Typography>
			<Button
				variant="contained"
				fullWidth
				color="primary"
				aria-label={"addBonus"}
				onClick={() =>
					props.addScoredBonus(props.bonus.name, props.bonus.id)
				}
				data-testid={"button-bonus-" + props.bonus.id}
			>
				Any
			</Button>
		</Paper>
	)
