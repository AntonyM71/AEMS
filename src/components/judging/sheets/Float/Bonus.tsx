import Button from "@material-ui/core/Button"
import Paper from "@material-ui/core/Paper"
import Typography from "@material-ui/core/Typography"
import React from "react"
import { useStyles } from "../../../../style/Styles"
import { BonusPropsType } from "./Interfaces"

export const Bonus = (props: BonusPropsType) => {
	const classes = useStyles()

	return (
		<Paper className={classes.moveBox}>
			<Typography>{props.bonus.name}</Typography>
			<Button
				variant="contained"
				fullWidth
				color="primary"
				aria-label={"addBonus"}
				onClick={() =>
					props.addScoredBonus(props.bonus.name, props.bonus.id)
				}
			>
				Any
			</Button>
		</Paper>
	)
}
