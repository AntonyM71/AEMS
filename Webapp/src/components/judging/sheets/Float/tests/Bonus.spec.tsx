import React from "react"
import { fireEvent, render } from "@testing-library/react"
import { Bonus } from "../Bonus"
import { BonusPropsType } from "../Interfaces"

describe("Bonus", () => {
	const addScoredMoveSpy = jest.fn()
	const addScoredBonusSpy = jest.fn()
	const BonusProps: BonusPropsType = {
		bonus: {
			id: "1234",
			name: "Air",
			shortName: "A"
		},
		addScoredMove: addScoredMoveSpy,
		addScoredBonus: addScoredBonusSpy
	}
	it("matches the SnapShot", () => {
		const wrapper = render(<Bonus {...BonusProps} />)
		expect(wrapper).toMatchSnapshot()
	})
	it("calls addScoredBonus when clicked", () => {
		const wrapper = render(<Bonus {...BonusProps} />)
		const wantedCall = ["Air", "1234"]
		expect(addScoredBonusSpy).toHaveBeenCalledTimes(0)

		fireEvent.click(wrapper.getByLabelText("addBonus"), {
			button: 1
		})
		expect(addScoredBonusSpy).toHaveBeenCalledTimes(1)
		expect(addScoredBonusSpy).toHaveBeenCalledWith(...wantedCall)
	})
})
