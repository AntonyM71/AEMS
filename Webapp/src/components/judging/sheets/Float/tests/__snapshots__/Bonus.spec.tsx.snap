// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bonus matches the SnapShot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Air
        </p>
        <button
          aria-label="addBonus"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          data-testid="button-bonus-1234"
          tabindex="0"
          type="button"
        >
          Any
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Air
      </p>
      <button
        aria-label="addBonus"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        data-testid="button-bonus-1234"
        tabindex="0"
        type="button"
      >
        Any
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
